### PR TITLE
Fix/Embed - Fix for Duration query param getting removed

### LINKED ISSUE #6127: [CAL-633] /embed - Call duration is not set to the default
When I go to book a call in the embed, it does not preselect the default duration

I then end up with "undefined minutes"

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-633](https://linear.app/calcom/issue/CAL-633/embed-call-duration-is-not-set-to-the-default)</sub>

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

Fixes #6127

**Environment**: Staging(main branch) / Production

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [ ] Test A
- [ ] Test B

### CODE REVIEW

File: apps/web/components/booking/BookingDescription.tsx [hunk 1]
```diff
@@ -55,7 +55,7 @@ const BookingDescription: FC<Props> = (props) => {
         }
       }
     }
-  }, []);
+  }, [setDuration, eventType.length, eventType.metadata?.multipleDuration, duration]);
```
Comment (hariombalhara) [inline apps/web/components/booking/BookingDescription.tsx:58]: This works till I figure out why duration param is being removed.

Comment (hariombalhara) [reply apps/web/components/booking/BookingDescription.tsx:58]: Using shallow:false on `useRouterQuery` also fixes the issue.

**Review by PeerRich — APPROVED** [review]

File: apps/web/components/booking/BookingDescription.tsx (hunk 1 above)
Comment (PeerRich) [reply apps/web/components/booking/BookingDescription.tsx:58]: so can we merge this?

Comment (hariombalhara) [reply apps/web/components/booking/BookingDescription.tsx:58]: Yeah, it's safe to merge and also fixes the issue.

But the root cause for why it happens only in embed isn't clear.

**Pull request merged**